extensions/hibernate-orm/pom.xml

```diff
@@ -44,24 +44,13 @@
                                     <!-- The jdk-reflection is not yet something we can avoid -->
                                     <!--<bundledSignature>jdk-reflection</bundledSignature>-->
 
-                                    <!-- All following signatures should be replicated for each target JDK version we intend to support -->
-                                    <bundledSignature>jdk-unsafe-11</bundledSignature>
-                                    <bundledSignature>jdk-unsafe-12</bundledSignature>
-                                    <bundledSignature>jdk-unsafe-13</bundledSignature>
-                                    <bundledSignature>jdk-unsafe-14</bundledSignature>
-                                    <bundledSignature>jdk-unsafe-15</bundledSignature>
-
-                                    <bundledSignature>jdk-deprecated-11</bundledSignature>
-                                    <bundledSignature>jdk-deprecated-12</bundledSignature>
-                                    <bundledSignature>jdk-deprecated-13</bundledSignature>
-                                    <bundledSignature>jdk-deprecated-14</bundledSignature>
-                                    <bundledSignature>jdk-deprecated-15</bundledSignature>
-
-                                    <bundledSignature>jdk-internal-11</bundledSignature>
-                                    <bundledSignature>jdk-internal-12</bundledSignature>
-                                    <bundledSignature>jdk-internal-13</bundledSignature>
-                                    <bundledSignature>jdk-internal-14</bundledSignature>
-                                    <bundledSignature>jdk-internal-15</bundledSignature>
+                                    <!--
+                                      This will automatically choose the right
+                                      signatures based on 'maven.compiler.release':
+                                    -->
+                                    <bundledSignature>jdk-unsafe</bundledSignature>
+                                    <bundledSignature>jdk-deprecated</bundledSignature>
+                                    <bundledSignature>jdk-internal</bundledSignature>
```

Comment:
I'm not sure about this change. I don't remember why all these versioned signatures were introduced.

@Sanne @yrodiere rings any bell?

Reply:
I think the original comment is correct: "All following signatures should be replicated for each target JDK version we intend to support", granted it seems it hasn't been updated to include 16 - 20.

But testing just for the JDK which happens to be used for maven.compiler.release would be less strict; I don't see any reason to do that?

Reply:
Great job to avoid unnecessary dependencies to core though! Having jars in the bootstrap classloader makes the app unnecessarily fat, as it can't be optimised like we do for regular dependencies.

Reply:
> Using versioned signatures is problematic since a new API method can be introduced in a new JDK version

But that's exactly why we want to test them all; Quarkus might "target" Java 11 but it's also meant to be compatible with any later versions, at leat the ones stable at the time of building a certain release.

Reply:
@jorsol awesome :+1:  thanks!